client/src/components/CompaniesTable.tsx

```diff
@@ -0,0 +1,85 @@
+import { DataGrid, GridColDef, GridToolbar } from "@mui/x-data-grid";
+import CompanyTableFooter from "../components/CompanyTableFooter";
+import { useState } from "react";
+
+type Company = {
+  id: number;
+  name: string;
+  address: string;
+  krs: string;
+};
+
+declare module "@mui/x-data-grid" {
+  interface FooterPropsOverrides {
+    companyId: number | null;
+  }
+}
+
+const columns: GridColDef[] = [
+  { field: "id", headerName: "ID", width: 70 },
+  { field: "name", headerName: "Company name", width: 250 },
+  { field: "address", headerName: "Address", width: 250 },
+  { field: "krs", headerName: "KRS", width: 250 },
+];
+
+const rows: Company[] = [
+  {
+    id: 1,
+    name: "Revolve Healthcare",
+    address: "Katowice, ul. Porcelanowa 23 40-246",
+    krs: "0000972657",
+  },
+  {
+    id: 2,
+    name: "Neubloc Polska",
+    address: "Katowice, ul. Grabowa 2 40-172",
+    krs: "0000335382",
+  },
+];
+
+function findRow(id: number): Company | null {
+  for (const row of rows) {
+    if (row.id === id) {
+      return row;
+    }
+  }
+  return null;
```

Comment:
to samo napisane deklaratywnie:
```suggestion
 return rows.find(r=>r.id===id);
```

Tu sobie poczytaj o przydatnych metodach na instancjach Array https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array
(find, forEach, map, filter, reduce, some, every)